sources/core/Stride.Core.AssemblyProcessor/ComplexSerializerRegistry.cs

```diff
@@ -282,25 +281,33 @@ public static IEnumerable<SerializableItem> GetSerializableItems(TypeDefinition
             var fields = new List<FieldDefinition>();
             var properties = new List<PropertyDefinition>();
 
-            var fieldEnum = type.Fields.Where(x => (x.IsPublic || (x.IsAssembly && x.CustomAttributes.Any(a => a.AttributeType.FullName == "Stride.Core.DataMemberAttribute"))) && !x.IsStatic && !ignoredMembers.Contains(x));
+            foreach (var field in type.Fields)
+            {
+                if (field.IsStatic)
+                    continue;
 
-            // If there is a explicit or sequential layout, sort by offset
-            if (type.IsSequentialLayout || type.IsExplicitLayout)
-                fieldEnum = fieldEnum.OrderBy(x => x.Offset);
+                if (ignoredMembers.Contains(field))
+                    continue;
 
-            foreach (var field in fieldEnum)
-            {
-                fields.Add(field);
+                if (field.IsPublic || (field.IsAssembly && field.CustomAttributes.Any(a => a.AttributeType.FullName == "Stride.Core.DataMemberAttribute")))
```

Comment:
this would break user code that has a inherited version of the datamemberattribute

lets say i make a class `public class FancyDataMember : DataMemberAttribute`
then that would be ignored even though its still a datamember attribute


in roslyn i would get from the compilation the type and check if its original definition is from datamemberattribute and if so it counts ie
```cs
    public static bool HasAttribute(this ISymbol symbol, INamedTypeSymbol attribute)
    {
        if (symbol.GetAttributes().Any(attr => attr.AttributeClass?.OriginalDefinition.Equals(attribute, SymbolEqualityComparer.Default) ?? false))
        {
            return true;
        }
        return false;
    }
```
i dont know if something like that is possible in cecil

but from looking at the current implementation that case isnt handled either.. so it would be a bug fix too in line 311

Reply:
line 311 of the old code

Reply:
It's actually a fairly involved change, types have to be resolved which likely incur a non-trivial perf' cost if used within this scope. There should be an initial phase to process all types to look for these, resolve only when necessary, and finally, pass those type refs to this method to filter.
And given how niche this issue would be, best to leave this for when we rework the assembly processor.

Reply:
I don't think we can support custom subclasses of attributes. That's not a common practice and most analyzers would not care as they resolve symbols explicitly.

Reply:
allright, then we can leave it as is